commit 7d545332c0c1f9879f805a23ae66f11865521f3d
Author: Reid Linnemann <rlinnemann@netgate.com>
Date:   Thu Aug 28 17:09:37 2025 +0000

    Only delete a local user on rpc sync if not being modified. Fixes #16391

diff --git a/src/etc/inc/auth.inc b/src/etc/inc/auth.inc
index 752c23c886..1acd6b2eab 100644
--- a/src/etc/inc/auth.inc
+++ b/src/etc/inc/auth.inc
@@ -506,7 +506,13 @@ function local_sync_accounts($u2add, $u2del, $g2add, $g2del) {
 			log_error(sprintf(gettext("Running: %s"), $cmd));
 		}
 		mwexec($cmd);
-		local_group_del_user($user);
+		foreach ($u2add as $adduser) {
+			if ($adduser['id'] == $user['id']) {
+				// don't delete a user from the local users if it is actually being modified
+				break;
+			}
+			local_group_del_user($user);
+		}
 
 		for ($i = 0; $i < count($users); $i++) {
 			if ($users[$i]['name'] == $user['name']) {
